src/Upgrade/Application/Strategy/Composer/Fixer/FeatureDevMasterPackageFixerStep.php

```diff
@@ -0,0 +1,91 @@
+<?php
+
+/**
+ * Copyright © 2016-present Spryker Systems GmbH. All rights reserved.
+ * Use of this software requires acceptance of the Evaluation License Agreement. See LICENSE file.
+ */
+
+declare(strict_types=1);
+
+namespace Upgrade\Application\Strategy\Composer\Fixer;
+
+use Upgrade\Application\Adapter\PackageManagerAdapterInterface;
+use Upgrade\Application\Dto\StepsResponseDto;
+use Upgrade\Domain\Entity\Collection\PackageCollection;
+use Upgrade\Domain\Entity\Package;
+
+class FeatureDevMasterPackageFixerStep extends AbstractFeaturePackageFixerStep
+{
+    /**
+     * @var string
+     */
+    public const ALIAS_DEV_MASTER = 'dev-master as 202000.0';
+
+    /**
+     * @var bool
+     */
+    protected bool $isFeatureToDevMasterEnabled;
+
+    /**
+     * @param \Upgrade\Application\Adapter\PackageManagerAdapterInterface $packageManager
+     * @param bool $isFeatureToDevMasterEnabled
+     */
+    public function __construct(PackageManagerAdapterInterface $packageManager, bool $isFeatureToDevMasterEnabled = false)
+    {
+        $this->isFeatureToDevMasterEnabled = $isFeatureToDevMasterEnabled;
+
+        parent::__construct($packageManager);
+    }
+
+    /**
+     * @param \Upgrade\Application\Dto\StepsResponseDto $stepsExecutionDto
+     *
+     * @return bool
+     */
+    public function isApplicable(StepsResponseDto $stepsExecutionDto): bool
+    {
+        if (!$this->isFeatureToDevMasterEnabled) {
+            return false;
+        }
+
+        return parent::isApplicable($stepsExecutionDto);
+    }
+
+    /**
+     * @param \Upgrade\Application\Dto\StepsResponseDto $stepsExecutionDto
+     *
+     * @return \Upgrade\Application\Dto\StepsResponseDto
+     */
+    public function run(StepsResponseDto $stepsExecutionDto): StepsResponseDto
+    {
+        $messages = $stepsExecutionDto->getOutputMessages();
+        $foundMessages = (array)preg_grep(static::PATTERN, $messages);
```

Comment:
move `$messages` and `$foundMessages` before the loop where it's used

Reply:
I need to have what we have before here